CriticalMass/IDStore.swift

```diff
@@ -0,0 +1,22 @@
+//
+//  IDStore.swift
+//  CriticalMaps
+//
+//  Created by Leonard Thomas on 5/2/19.
+//
+
+import Foundation
+
+public class IDStore: IDProvider {
+    public init(currentDate: Date = Date()) {
+        let deviceID = UIDevice.current.identifierForVendor?.uuidString ?? UUID().uuidString
+
+        let format = DateFormatter()
+        format.dateFormat = "yyyy-MM-dd"
+        let dateString = format.string(from: currentDate)
+
+        id = (deviceID + dateString).md5
+    }
```

Comment:
Looks good to me. Note that in android we add 6 hours to the current time to shift the date to the next day. This way on the day of a CM (and the mass goes on until after 12pm) the ID is constant in case someone restarts the app after 12pm. tbh this is just a minor edgecase though.

Reply:
How about events that start earlier? Wouldn’t it be better to check when the last session has started and if it’s more than 24 hours, renew the id without looking for the exact date